### ISSUE TITLE
Bug: Conflict rules in TypeScript on whether or not to have a semicolon after an `export default interface`

### ISSUE THREAD

**Issue by thisrabbit** [eslint/eslint#16200]
### Environment

Node version: v16.13.2
npm version: 8.17.0
Local ESLint version: v8.21.0
Global ESLint version: (Not appliable)
Operating System: Windows 10 / macOS

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "env": {
    "es2021": true,
    "jest/globals": true,
    "node": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 12,
    "sourceType": "module"
  },
  "plugins": [
    "@typescript-eslint",
    "jest"
  ]
}

```
</details>

```ts
export default interface MatchResult {
  foo: 'aaa';
}
```

### What did you expect to happen?

As other TS rules did, code style should conform to `@typescript-eslint/no-extra-semi`.

### What actually happened?

**When not inserting a semicolon:** An `ESLint: Missing semicolon.(semi)` throws;

**When inserting a semicolon:** Both `ESLint: Missing semicolon.(semi)` and `ESLint: Unnecessary semicolon.(@typescript-eslint/no-extra-semi)` throws.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

`interface` is some of the TypeScript things that can be exported as the default, this may be a corner case.

**Comment by mdjermanovic** [comment]
Hi @thisrabbit, thanks for the issue!

The core `semi` rule does not support TypeScript code. You should use the [`@typescript-eslint/semi`](https://typescript-eslint.io/rules/semi/) rule instead.

```json
{
  "semi": "off",
  "@typescript-eslint/semi": ["error"]
}
```
